packages/atomic/src/components/result-template-components/result-template-decorators.tsx

```diff
@@ -85,3 +85,79 @@ export const resultContext = (element: Element) =>
       reject(new MissingResultParentError(element.nodeName.toLowerCase()));
     }
   });
+
+/**
+ * A [StencilJS property decorator](https://stenciljs.com/) to be used for folded result template components.
+ * This allows the Stencil component to fetch the current folded result from its rendered parent, the `atomic-result` component.
+ *
+ *
+ * @example
+ * @FoldedResultContext() private foldedResult!: FoldedResult;
+ *
+ * For more information and examples, view the "Utilities" section of the readme.
+ */
+export function FoldedResultContext() {
```

Comment:
Similar to my above comment, it might be possible to modify `ResultContext` decorator to always resolve a `FoldedResult`, with an empty `children` array when there is no folding active.

So that we would not need two different decorator.

I'm not 100% certain it's possible, just a suggestion